syntax = "proto2";
package ndn_message;

// Protobuf has no "outer" message type, so we need to put the TLV RepoCommandParameter
// message inside an outer "typeless" RepoCommandParameterMessage.

message RepoCommandParameterMessage {
  message Name {
    repeated bytes component = 8;
  }

  message RepoCommandParameter {
    optional Name name = 7;
    optional uint64 start_block_id = 204;
    optional uint64 end_block_id = 205;
    optional uint64 process_id = 206;
    optional uint64 max_interest_num = 211;
    optional uint64 watch_timeout = 212;
    optional uint64 watch_status = 213;
    optional uint64 interest_lifetime = 214;
  }

  // A RepoCommandParameterMessage has one RepoCommandParameter.
  required RepoCommandParameter repo_command_parameter = 201;
}